File: 2019/Day 04/try.hs
import Text.Printf
import Control.Applicative
import Data.List

main :: IO()
main = do
  let fn = liftA2 (&&) (nub >>= (/=)) (sort >>= (==))
  let xs = filter fn $map show [136760..595730]
  printf "Silver: %d\n" $length xs
  printf "Gold: %d\n" $length $filter
    (elem 2.map length.group) xs

File: 2019/Day 04/try.rb
part1 = (136760..595730)
  .map(&:to_s).map(&:chars)
  .select{|xs| xs.sort == xs}
  .map{|xs| xs.slice_when{
    |a, b| a != b}.map(&:size)
  }.reject{|xs| xs.max < 2}

part2 = part1.filter{|xs|
  xs.include? 2}

puts "Silver: #{part1.size}",
  "Gold: #{part2.size}"
